### PR TITLE
feat: implement get user task var - controller

### LINKED ISSUE #22903: Implement `Get Variable for User Task`
### Description

Variables endpoint providing search for variables.

* The endpoints should be implemented as described in [REST API Developer Guide](https://github.com/camunda/camunda/blob/main/docs/rest-controller.md)
* Variable objects are defined in
  * [Variable](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.i5jd4xzabhje) 
 
### Endpoint

* `GET /v2/user-tasks/{userTaskKey}/variables`
* [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.sc4u7eavwp5g)

### Acceptance Criteria

- [ ] REST API endpoint incl. OpenAPI spec
- [ ] Java client support
- [ ] Integration tests

```[tasklist]
### Subtasks
- [ ] https://github.com/camunda/camunda/issues/23558
```

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

- Get User Task Variables Controller 

The Aggregator is not included to this implementation, it will be an increment in a separate PR/Issue

Issue for Aggregator: https://github.com/camunda/camunda/issues/23558 

Issue: 
https://github.com/camunda/camunda/issues/22903

### CODE REVIEW

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/UserTaskQueryController.java [hunk 1]
```diff
@@ -107,4 +122,52 @@ public ResponseEntity<Object> getFormByUserTaskKey(
       return RestErrorMapper.mapProblemToResponse(problemDetail);
     }
   }
+
+  @PostMapping(
+      path = "/{userTaskKey}/variables",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<Object> searchVariables(
```
Comment (ralfpuchert) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/UserTaskQueryController.java:130]: Can we use the actual type `ResponseEntity<VariableSearchQueryResponse>` here?

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/UserTaskQueryController.java [hunk 2]
```diff
@@ -107,4 +122,52 @@ public ResponseEntity<Object> getFormByUserTaskKey(
...
+        (treePath != null && !treePath.isEmpty())
+            ? Arrays.stream(treePath.split("/")).map(Long::valueOf).collect(Collectors.toList())
+            : Collections.emptyList();
+
+    return SearchQueryRequestMapper.toUserTaskVariableQuery(
+            userTaskVariablesSearchQueryRequest, treePathList)
+        .fold(RestErrorMapper::mapProblemToResponse, this::searchUserTaskVariableQuery);
+  }
+
+  private ResponseEntity<Object> searchUserTaskVariableQuery(final VariableQuery query) {
```
Comment (ralfpuchert) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/UserTaskQueryController.java:151]: Same as before.

**Review by ralfpuchert — APPROVED** [review]
LGTM. 2 minor questions/comments about type declaration :)

**Review by pepopowitz — COMMENTED** [review]
Spec changes look good from my end 👍

**Pull request merged**